usr/share/rear/rescue/default/500_ssh.sh

```diff
@@ -54,5 +57,110 @@ if has_binary sshd; then
         *     ) echo "root:$(echo $SSH_ROOT_PASSWORD | openssl passwd -1 -stdin):::::::" > $ROOTFS_DIR/etc/shadow ;;
         esac
     fi
-fi
 
+    # Part 2: SSH client (ssh) - this is for accessing other systems (e.g. the backup server) via SSH from the recovery system
+
+    Log "Copying SSH client configuration"
+
+    local target_dir="$ROOTFS_DIR/root/.ssh" original_file target_file passphrase_set="no" passphrase="" repeated_passphrase
+
+    mkdir -p -m u=rwx,go=- "$target_dir"
+
+    for original_file in /root/.ssh/id_* /root/.ssh/known_host[s]; do
+        case "$original_file" in
+            (*/id_*.pub|*/known_hosts)
+                # Public key files or 'knownhosts' contain no secrets and can be copied as-is
+                Log "Copying $original_file"
+                cp -p "$original_file" "$target_dir"
+                ;;
+
+            (*/id_*)
+                # Private key files will be copied, examined and, if desired, passphrase-protected
+
+                target_file="$target_dir/$(basename "$original_file")"
+                cp -p "$original_file" "$target_file"
```

Comment:
In both cases you copy the file. In the case of a key file you check it an optionally remove it.

I think it would be even simpler to first copy the entire `/root/.ssh` directory as it is and then iterate over the private key files and delete those that are not compliant. Less code, less `case` and less complexity.

Reply:
The idea was to copy well-known files only. If the admin leaves other security-related files in this well-protected directory, we'd still be safe. Same goes with new key file types which might come up in the future. Better not copy anything that we don't really know about to an insecure rescue medium.

Reply:
I would copy all of the SSH config directory and fix problems when they arise. Typically there is also some delay between new features showing up and between those features being used.

In any case, you could collapse that all into a single `cp` command with multiple source arguments that represent your whitelist of known files to copy.

Reply:
Considered that, too. I'd still end up with a case to deal with private key files matching `id_*` but not `id_*.pub`. See my upcoming commit if you'd just consider it good enough.